tests/integration/queryparam.spec.ts

```diff
@@ -0,0 +1,152 @@
+// tslint:disable-next-line:no-require-imports no-implicit-dependencies
+import express = require('express');
+
+import {
+    Controller,
+    ApiController,
+    Dino,
+    HttpGet,
+    QueryParam,
+    toInteger,
+    toNumber,
+    toBoolean,
+    request
+} from '../index';
+
+@Controller('/testcontroller')
+class TestControllerFake extends ApiController {
+    @HttpGet('/toBoolean')
+    toBoolean(@QueryParam(toBoolean) val: boolean): any {
+        return { data: val };
+    }
+    @HttpGet('/toInteger')
+    toInteger(@QueryParam(toInteger) val: number): any {
+        return { data: val };
+    }
+    @HttpGet('/toNumber')
+    toNumber(@QueryParam(toNumber) val: number): any {
+        return { data: val };
+    }
+
+    @HttpGet('/toBooleanPathAndQuery/:val')
+    toBooleanPathAndQuery(@QueryParam(toBoolean) val: number): any {
+        return { data: val };
+    }
+
+    @HttpGet('/toIntegerPathAndQuery/:val')
+    toIntegerPathAndQuery(@QueryParam(toInteger) val: number): any {
+        return { data: val };
+    }
+
+    @HttpGet('/toNumberPathAndQuery/:val')
+    toNumberPathAndQuery(@QueryParam(toNumber) val: number): any {
```

Comment:
// GET '/toNumberWithoutQueryParam/45?val=50
@HttpGet('/toNumberWithoutQueryParam/:val')
toNumberPathAndQuery(@Parse(toNumber) val: number): any { }

Could you please confirm what value is injected into val? 
It should pick it from variable segments when we use @Parse and from queryParams when we use @QueryParam. 
I am away from my machine, I will test this once I get access.

Reply:
Yes.